lib/hsa/mcwamp_hsa.cpp

```diff
@@ -119,6 +119,7 @@ int HCC_MAX_QUEUES = 20;
 #define HCC_PROFILE_SUMMARY (1<<0)
 #define HCC_PROFILE_TRACE   (1<<1)
 int HCC_PROFILE=0;
+int HCC_FLUSH_ON_WAIT=1;
```

Comment:
How about defining the HCC_FLUSH_ON_WAIT as a count of asyncops to reclaim, so it can be tuned for performance analysis? A value of 0 or undefined env would be equivalent to no reclaim.

Reply:
This is possible, sure.  Is it desirable to not recover all resources?

I imagine we would want to recover the resources associated with the oldest HSAOps first.  We do not maintain an index for the oldest op.

Reply:
Isn't the worst case cleaning up HCC_ASYNCOPS_SIZE (16k) items? The env variable as a count to reclaim could be used as an upper limit to limit any intermittent/variable performance impact due to pending ops to, say, a more "predictable" behavior. Also, could be used to tune and collect data for analysis.

Reply:
I'd suggest the HCC_ASYNCOPS_SIZE being the first parameter to tune.  I think the current default is too large and there may be a sweet spot where it's sufficient large to accomodate a relatively large batch of dispatches without forcing a wait.

Reply:
if hsa signals are the precious resource, perhaps we should make that the limit.  also, #1173 could help reduce the resource use, but it is a much bigger than than this current PR.